src/js/frontend/src/components/login/Login.tsx

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import { GoogleLogin } from "@react-oauth/google";
+import * as Yup from "yup";
+import { Field, Form, Formik } from "formik";
+
+import { signInWithGoogleService } from "../../services/login/signInWithGoogleService";
+import { loginService } from "../../services/login/loginService";
+
+const Login = () => {
+  const [state, setState] = React.useState({
+    loginFailed: false,
+  });
+
+  const validationSchema = Yup.object().shape({
+    userName: Yup.string().required("Username Required"),
+    password: Yup.string().required("Password Required"),
+  });
+
+  return (
```

Comment:
This code block is really large and complicated. Can we somehow break it up? Like move the GoogleLogin 'on success' code to a function, same with the'onSubmit' callback. A comment on the 'loginFailed' that describes when it changes from true -> false or false -> true would be helpful as well.